lib/fizzy/execute.cpp

```diff
@@ -138,16 +138,23 @@ inline bool store_into_memory(
 inline uint32_t grow_memory(
     bytes& memory, uint32_t delta_pages, uint32_t memory_pages_limit) noexcept
 {
+    // Guaranteed by instantiate allocation + resize in grow_memory.
+    assert(memory.size() % PageSize == 0);
     const auto cur_pages = memory.size() / PageSize;
-    assert(cur_pages <= size_t(std::numeric_limits<int32_t>::max()));
-    const auto new_pages = cur_pages + delta_pages;
-    assert(new_pages >= cur_pages);
+    // Guaranteed by instantiate allocation + resize in grow_memory.
+    assert(cur_pages <= MemoryPagesValidationLimit);
```

Comment:
Probably `MemoryPagesValidationLimit` should be renames, because it will be used not only for validation.